src/main/java/teammates/ui/controller/InstructorHomeCourseAjaxPageData.java

```diff
@@ -0,0 +1,166 @@
+                                       Const.Tooltips.COURSE_DETAILS);
+        
+        ElementTag edit = createButton("Edit",
+                                       className + "edit-for-test",
+                                       getInstructorCourseEditLink(courseId),
+                                       Const.Tooltips.COURSE_EDIT);
+        
+        ElementTag add = createButton("Add Session",
+                                      className + "add-eval-for-test",
+                                      getInstructorFeedbacksLink(courseId),
+                                      Const.Tooltips.COURSE_ADD_FEEDBACKSESSION);
+        addAttributeIf(!instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION),
+                       add, disabled, disabled);
+        
+        ElementTag archive = createButton("Archive",
+                                          className + "archive-for-test",
+                                          getInstructorCourseArchiveLink(courseId, true, true),
+                                          Const.Tooltips.COURSE_ARCHIVE);
+        addAttributeIf(true, archive, "onclick", "return toggleArchiveCourseConfirmation('" + courseId + "')");
+        
+        ElementTag delete = createButton("Delete",
+                                         className + "delete-for-test",
+                                         getInstructorCourseDeleteLink(courseId, true),
+                                         Const.Tooltips.COURSE_DELETE);
+        addAttributeIf(!instructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COURSE),
+                       delete, disabled, disabled);
+        addAttributeIf(true, delete, "onclick", "return toggleDeleteCourseConfirmation('" + courseId + "')");
+        
+        if (pendingCommentsCount <= 0) {
+            return Arrays.asList(enroll, view, edit,add, archive, delete);
+        } else {
+            String pendingGraphic = "<span class=\"badge\">" + pendingCommentsCount + "</span>"
+                                    + "<span class=\"glyphicon glyphicon-comment\"></span>"
+                                    + "<span class=\"glyphicon glyphicon-arrow-right\"></span>"
+                                    + "<span class=\"glyphicon glyphicon-envelope\"></span>";
+            ElementTag pending = createButton(
+                    pendingGraphic,
+                    className + "notify-pending-comments-for-test",
+                    getInstructorStudentCommentClearPendingLink(courseId),
+                    "Send email notification to recipients of " + pendingCommentsCount
+                        + " pending " + (pendingCommentsCount > 1 ? "comments" : "comment"));
+    
+            return Arrays.asList(enroll, view, edit,add, archive, pending, delete);
+        }
+    }
+    
+    private List<Map<String, Object>> createSessionRows(List<FeedbackSessionAttributes> sessions,
```

Comment:
Yes, I do want to refactor the instructor home files, they were the first ones done, so they didn't benefit from the experience from the rest of the code written during the JSTL migration.

Should I make an issue for it? "Refactor InstructorHome" or something?